File: backend-main/src/main/java/com/cosmetic/gg/service/supply/SupplierService.java
package com.cosmetic.gg.service.supply;

import java.util.List;

import com.cosmetic.gg.common.enums.EStatus;
import com.cosmetic.gg.common.model.Error;
import com.cosmetic.gg.entity.supply.Supplier;

public interface SupplierService {

	List<Supplier> search(String keyword, EStatus status);
	
	List<Error> validator(Supplier supplier);
	
	Supplier create(Supplier supplier);
	
	Supplier update(Supplier supplier);
	
	Error delete(String id);
	
	Supplier getById(String id);
}

File: fontend-main/src/app/core/services/supplier.service.ts
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, catchError, map, of } from 'rxjs';
import { HttpClientService } from './httpClient.service';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class SupplierService {
  httpOptions = {
    headers: new HttpHeaders({
      'Content-Type': 'application/json',
    })
  };
  constructor(private http: HttpClient, private httpService: HttpClientService){}

  //#region CRUD
  search(data: any): Observable<any> {
    let httpParams = new HttpParams();
    Object.keys(data).forEach(function (key) {
      httpParams = httpParams.append(key, data[key]);
    });
    const httpOptions = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
      }),
      params: httpParams
    };
    return this.http.get(`${environment.baseUrl}${environment.basePath}/supplier`, httpOptions);
  }

  create(data: any): Observable<any> {
    return this.http.post(`${environment.baseUrl}${environment.basePath}/supplier`, data);
  }

  update(id: string, data: any): Observable<any> {
    return this.http.put(`${environment.baseUrl}${environment.basePath}/supplier/` +  id, data, this.httpOptions);
  }

  delete(id: string): Observable<any> {
    return this.http.delete(`${environment.baseUrl}${environment.basePath}/supplier/` + id);
  }

  getById(id: string): Observable<any> {
    return this.http.get(`${environment.baseUrl}${environment.basePath}/supplier/` + id);
  }
  //#endregion
}
